docs/testplan/latency_test.md

```diff
@@ -0,0 +1,99 @@
+# SONiC Switch Latency Test
+
+- [SONiC Switch Latency Test](#sonic-switch-latency-test)
+  - [Test Objective](#test-objective)
+  - [Test Setup](#test-setup)
+  - [Test Case 1: Full-mesh Latency Test](#test-case-1-full-mesh-latency-test)
+  - [Test case 2: Latency Test with Oversubscribed Port](#test-case-2-latency-test-with-oversubscribed-port)
+  - [Metrics Processing](#metrics-processing)
+
+## Test Objective
+
+This test aims to measure the latency introduced by a switch under the fully loaded condition.
+
+## Test Setup
+
+> **Notice**: Currently, we are testing exclusively in store-and-forward mode.
+
+1. **Maximize Traffic Load on DUT**
+
+   This test uses all traffic generator ports defined in the test topology to generate the maximum possible traffic volume, stressing the DUT to capture its latency behavior under full load.
+
+   - **Switch wiring**: All DUT ports must be connected to traffic generators—either directly or through intermediary devices.
+
+2. **Establish BGP Sessions and Routes**
+
+   Follow the steps outlined in the multi-tier BGP test plan to:
+   - Set up the necessary IPv6 BGP sessions
+   - Advertise the desired IPv6 routes
+
+3. **Configure Traffic Flows**
+
+   Set up traffic items on the traffic generators to ensure:
+   - Every DUT port receives traffic
+   - Every route on the DUT is exercised
+
+   For example:
+   - For each of the four traffic generators, define two topologies, each containing four **physical** ports.
+   - Configure and enable Layer 2 and Layer 3 protocols for both topologies.
+   - In IXIA-1, define two uni-directional traffic items:
+     - One targeting IXIA-2
+     - One targeting IXIA-4
+     - Both set to full mesh mode
+     - Each at 60% line rate, with a packet size of 1024 bytes
+
+   - Similarly, in IXIA-2, define traffic items targeting IXIA-1 and IXIA-3.
+   - Repeat the pattern across all traffic generators to establish a robust traffic matrix.
+
+## Test Case 1: Full-mesh Latency Test
+
+1. **Baseline Latency Measurement**
+   - Start all the traffic items simultaneously and run them for 1 minute. Record latency statistics.
+   - Record latency statistics from each traffic item.
+   - Compare latency differences between the two traffic items sourced from each traffic generator.
+2. **Packet Size Variation**
+   - Repeat the baseline test using different frame sizes: 64 bytes, 1024 bytes, 4096 bytes, and 8192 bytes.
```

Comment:
All the tests needs to support these variations, we should define a matrix or list before showing the test cases, saying all test cases should support the following variations.